sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -72,4 +75,55 @@ export class EducationProgramOfferingController {
       offeringDelivered: offering.offeringDelivered,
     }));
   }
+
+  @HasLocationAccess("locationId")
+  @Get("location/:locationId/education-program/:programId/offering/:offeringId")
+  async getProgramOffering(
+    @Param("locationId") locationId: number,
+    @Param("programId") programId: number,
+    @Param("offeringId") offeringId: number,
+  ): Promise<ProgramOfferingDto> {
+    //To retrive Education program offering corresponding to ProgramId and LocationId
+    const offering = await this.programOfferingService.getProgramOffering(
+      locationId,
+      programId,
+      offeringId,
+    );
+    if (!offering) {
+      throw new UnprocessableEntityException(
+        "Not able to find a Education Program Offering associated with the current Education Program, Location and offering.",
+      );
+    }
+    return offering;
+  }
+
+  @HasLocationAccess("locationId")
+  @Patch(
+    "location/:locationId/education-program/:programId/offering/:offeringId",
+  )
+  async updateProgramOffering(
+    @Body() payload: UpdateEducationProgramOfferingDto,
+    @Param("locationId") locationId: number,
+    @Param("programId") programId?: number,
+    @Param("offeringId") offeringId?: number,
+  ): Promise<number> {
+    const updatingResult = await this.formService.dryRunSubmission(
+      FormNames.EducationProgramOffering,
+      payload,
+    );
+
+    if (!updatingResult.valid) {
+      throw new UnprocessableEntityException(
+        "Not able to a update a program offering due to an invalid request.",
+      );
+    }
+
+    const updateProgramOffering = await this.programOfferingService.updateEducationProgramOffering(
+      locationId,
+      programId,
+      offeringId,
+      payload,
+    );
+    return updateProgramOffering.affected;
```

Comment:
So far, I believe that we are not returning anything for Patch/Put.

Reply:
Yeah we are not returning, but this is a good practice to return the updated rows and error handle if it fails. I did the same for InsitutionLocation patch

Reply:
The affect rows are supposed to be always 1 and the HTTP status is supposed to return the status os the operation.
When you mention that it is a good practice to return in this way, is there any reference that supports it?